File: src/main/java/com/github/javatrainingcourse/obogmanager/domain/repository/PasswordResetRequestRepository.java
/*
 * Copyright (c) 2017 mikan
 */

package com.github.javatrainingcourse.obogmanager.domain.repository;

import com.github.javatrainingcourse.obogmanager.domain.model.PasswordResetRequest;
import org.springframework.data.jpa.repository.JpaRepository;

/**
 * @author mikan
 * @since 0.1
 */
public interface PasswordResetRequestRepository extends JpaRepository<PasswordResetRequest, String> {
    PasswordResetRequest findByToken(String token);
}

File: src/main/kotlin/com/github/javatrainingcourse/obogmanager/domain/repository/PasswordResetRequestRepository.kt
/*
 * Copyright (c) 2017-2018 mikan
 */

package com.github.javatrainingcourse.obogmanager.domain.repository

import com.github.javatrainingcourse.obogmanager.domain.model.PasswordResetRequest
import org.springframework.data.jpa.repository.JpaRepository

/**
 * @author mikan
 * @since 0.1
 */
interface PasswordResetRequestRepository : JpaRepository<PasswordResetRequest, String> {
    fun findByToken(token: String): PasswordResetRequest?
}
